lib/rubocop/cop/lint/duplicate_require.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for duplicate `require`s and `require_relative`s.
+      #
+      # @example
+      #   # bad
+      #   require 'foo'
+      #   require 'bar'
+      #   require 'foo'
+      #
+      #   # good
+      #   require 'foo'
+      #   require 'bar'
+      #
+      class DuplicateRequire < Base
+        MSG = 'Duplicate `%<method>s` detected.'
+        REQUIRE_METHODS = %i[require require_relative].freeze
+
+        def_node_matcher :require_call?, <<~PATTERN
+          (send {nil? (const _ :Kernel)} {:#{REQUIRE_METHODS.join(' :')}} _)
+        PATTERN
+
+        def on_send(node)
+          return unless REQUIRE_METHODS.include?(node.method_name)
+
+          left_siblings_of(node).each do |sibling|
+            next unless require_call?(node) && duplicate_require?(node, sibling)
+
+            add_offense(node, message: format(MSG, method: node.method_name))
+          end
+        end
+
+        private
+
+        def left_siblings_of(node)
```

Comment:
There are a lot of places where such `siblings`, `left_siblings` are used. 
@marcandre Wdyt about adding such helpers to `rubocop-ast`?

Reply:
Agreed. rubocop-hq/rubocop-ast#93